lib/cyrusdb_twoskip.c

```diff
@@ -1548,7 +1540,7 @@ static int myfetch(struct dbengine *db,
     if (foundkey) *foundkey = db->loc.keybuf.s;
     if (foundkeylen) *foundkeylen = db->loc.keybuf.len;
 
-    if (!r && db->loc.is_exactmatch) {
```

Comment:
I mean, this is clearly correct, 'r' can't be non-zero here.  It's a pointless test.